library/Businessprocess/Modification/NodeRemoveAction.php

```diff
@@ -65,15 +65,19 @@ public function applyTo(BpConfig $config)
         $name = $this->getNodeName();
         $parentName = $this->getParentName();
         if ($parentName === null) {
-            $oldDisplay = $config->getBpNode($name)->getDisplay();
-            $config->removeNode($name);
-            if ($config->getMetadata()->isManuallyOrdered()) {
-                foreach ($config->getRootNodes() as $_ => $node) {
-                    $nodeDisplay = $node->getDisplay();
-                    if ($nodeDisplay > $oldDisplay) {
-                        $node->setDisplay($node->getDisplay() - 1);
-                    } elseif ($nodeDisplay === $oldDisplay) {
-                        break;  // Stop immediately to not make things worse ;)
+            if (! $config->hasBpNode($name)) {
```

Comment:
I wonder why that would be the case. I mean, a node without a parent must be a process node (root node), so if `hasBpNode()` returns false, is there actually a node at all? Does a `! $config->hasNode($name)` then suffice? Removal then is of course unnecessary.